### ISSUE TITLE
Using setuptools_scm + pbr

### ISSUE THREAD

**Issue by webknjaz** [pypa/setuptools-scm#170]
Hi,

I was curious whether I can use [`pbr`](https://docs.openstack.org/developer/pbr/) and `setuptools_scm` together. Here's a test `setup.py` module:
```python
#! /usr/bin/env python
"""package setuptools installer."""
from setuptools import setup


params = dict(
    setup_requires=['pbr==3.0.0', 'setuptools_scm>=1.15.0'],
    pbr=True,
    use_scm_version=True,
)

if __name__ == '__main__':
    setup(**params)
```

It turned out that technically it "works", but with a side effect: it looks like a method for version generation is picked randomly — one of above packages is picked for this purpose.
Just wanted to share/document my findings.

My motivation to try using "brand-new" package for version generation and "big-and-robust" one for other stuff (which is not supported by `setuptools_scm`, *yet?*).

Probably you may want to embed some check and alert users about unpredictable behavior if `pbr` is setup-required.

**Comment by RonnyPfannschmidt** [comment]
@webknjaz thanks for the heads-up - pbr has a history of doing too many things at once and being a massive bag of very special opinions

what do you mean by "big and robust" btw

**Comment by webknjaz** [comment]
@RonnyPfannschmidt I mean that it lets users keep parts of setup metadata in separate places doing it in a declarative way. I like being able to list dependencies in `requirements.txt`, most of metadata in `setup.cfg` and not keep some heavy stuff in `setup.py`. I think it's big amount of features. And I think that it's robust, because its widely used and supported by OpenStack, which has a huge ecosystem.

**Comment by jaraco** [comment]
> I like being able to list dependencies in requirements.txt

PyPA [recommends using install_requires for declaring the dependencies](https://caremad.io/posts/2013/07/setup-vs-requirement/) and referencing that from requirements.txt.

> most of metadata in setup.cfg

[Late versions of setuptools](http://setuptools.readthedocs.io/en/latest/history.html#v30-3-0) allow for declaration of most options in setup.cfg (possibly superseding features of pbr).

`setuptools_scm` is the blessed package for determining versions from SCM metadata. If third-party packages such as pbr run into conflict, those issues lie with those projects to reconcile the conficts. 

> it looks like a method for version generation is picked randomly

I suspect the order of operation is affected by one of two things: the order of dictionary keys of keyword arguments (which appears first in the dict, 'pbr' or 'use_scm_version') or maybe the presence of metadata (perhaps use_scm_version is picked up on the first build but subsequently pbr finds the version in the metadata and uses that).

In summary, I would suggest this is a won't fix and suggest instead that these packages can't work together or that pbr should be adapted to defer its version handling when setuptools_scm is present. If it were me, I would eschew pbr altogether and contribute to and rely on the official packaging tools.

**Comment by RonnyPfannschmidt** [comment]
@jaraco thanks for the comprehensive summary, it captures my gut feeling in eloquent wording
